go/libkb/sig_chain.go

```diff
@@ -705,61 +745,58 @@ func (l *SigChainLoader) AccessPreload() bool {
 }
 
 func (l *SigChainLoader) LoadLinksFromStorage() (err error) {
-	var curr LinkID
-	var links []*ChainLink
 	var mt *MerkleTriple
-	goodKey := true
 
 	uid := l.user.GetUID()
 
 	l.G().Log.CDebugf(l.ctx, "+ SigChainLoader.LoadFromStorage(%s)", uid)
 	defer func() { l.G().Log.CDebugf(l.ctx, "- SigChainLoader.LoadFromStorage(%s) -> %s", uid, ErrToOk(err)) }()
 
-	if mt, err = l.LoadLastLinkIDFromStorage(); err != nil || mt == nil {
+	if mt, err = l.LoadLastLinkIDFromStorage(); err != nil || mt == nil || mt.LinkID == nil {
```

Comment:
I kind of can't remember. But part of what's changed is that we used to do basically this check on line 761, and now this is the only place that does it.